app/components/chat_message/chat_message.js

```diff
@@ -35,4 +37,24 @@ export default class extends Controller {
     this.element.classList.remove('ChatMessage--expanded');
     this.toggleTarget.setAttribute('aria-expanded', 'false');
   }
+
+  copy() {
+    const text = this.textTarget.innerText;
+    const sender = this.data.get('sender-name');
+    let clipboardText = `${sender}: ${text}`;
+
+    if (!sender) {
+      clipboardText = text;
+    }
+
+    navigator.clipboard.writeText(clipboardText).then(() => this.onCopy());
+  }
+
+  onCopy() {
+    this.element.classList.add('ChatMessage--copied');
+
+    setTimeout(() => {
```

Comment:
Can't we implement that with CSS?

Reply:
Yes, that should be possible, e.g. using different transition delays – however I think that’d make it more opaque/more difficult to understand…